### ISSUE TITLE
Question mark in heading of extension manager UI looks like it should be clickable, but it isn't. Instead it shows extension path on hover

### ISSUE THREAD

**Issue by andrii-i** [jupyterlab/jupyterlab#14018]
## Description

Question mark in heading of extension manager UI looks like it should be clickable, but it isn't. Instead you have to hover over it and the information it shows (extension path) is not at all suggested by the ? mark at that location.

## Reproduce

1. Open Extension Manager
2. Note question mark in the heading 
![Screen Shot 2023-02-15 at 6 19 40 PM](https://user-images.githubusercontent.com/26686070/219261181-39a3363c-ef52-4815-aeff-6782e2b3d720.png)
![Screen Shot 2023-02-15 at 6 19 59 PM](https://user-images.githubusercontent.com/26686070/219261195-b4d48439-f6bd-4226-869d-59a08440aa23.png)

## Expected behavior

Question mark is used appropriately or not used at all.

## Context

Operating System and version: MacOS 12.6.3
Browser and version: Chrome version 109.0.5414.87 (Official Build) (arm64)
JupyterLab version: head-of-master (WIP 4.0.0)

**Comment by JasonWeill** [comment]
Should the "?" button be used for this information? It might be better put in another section.

**Comment by andrii-i** [comment]
I think there are 2 routes here:
1. If people who use extension manager need / want quick access to extension installation path: change question mark to be a button and show a popup with information about what is Extension Manager, that it allows you to quickly browse and install PyPi (https://pypi.org/) packages
2. If having installation path surfaced at such a level is not necessary: remove question mark, move extension installation path to some other section (I don't see an obvious place where to, any ideas?)

**Comment by GabrielaVives** [comment]
I don't fully understand what the installation path information is useful for. When and why do people need this information?

I agree that it must not be in this location, as it seems to be counter intuitive to all of you. Maybe for the moment we should remove this information until we have a better idea about where to display it?

**Comment by krassowski** [comment]
> I don't fully understand what the installation path information is useful for. When and why do people need this information?

This is needed because if a user installs a package into a wrong environment they may trigger wide ranging side-effects including a corruption of their JupyterLab installation (if that introduces a version conflict). The path will depend on how they configured their JupyterLab installation, and is needed because the path used by kernel may differ from the one used by JupyterLab itself causing confusion if not shown.

> Maybe for the moment we should remove this information until we have a better idea about where to display it?

Please no.

**Comment by GabrielaVives** [comment]
@krassowski as a user, do you have an idea about where you would expect to find this information?

**Comment by krassowski** [comment]
The full path is not super important (can be in a tooltip), the environment itself is. It could be exposed a little bit more by showing:
- environment name (conda and pyenv environments are named; other environments are usually referred to as system + version, e.g. "system 3.10")
- conda or Python icon depending if this is a conda environment or not; I am not sure if there are well-established icons for other environment managers

Long-term ideally someone will develop a "Conda extension manager" and it will be used automatically in place of "PyPI extension manager" when conda env is detected. However since no one expressed intent to maintain it in the core right now we have to work with what we have.

When I say conda I mean conda or mamba. Not sure if the distinction matters here.

**Comment by krassowski** [comment]
Maybe this could be a new section named "Environment: {env name}" which would be expanded when "Warning" is expended and collapsed once user accepted the warning?

Of note, we should check if the warning text needs updating before 4.0 release.

**Comment by andrii-i** [comment]
I have a couple of ideas, going from easiest to more complex. This could be implemented one after another and I think each step would be an improvement. @krassowski @GabrielaVives, what do you think?

1. (quick fix) Remove question mark as confusing, show environment path on hover over "PYPI MANAGER" (visibility does not change, less clutter/confusion)
2. Instead of question mark with installation path, show "environment name" and "conda or Python icon depending if this is a conda environment or not" (we could detect it based on the installation path itself or investigate more sophisticated detection methods). We could try fitting both instead of the question mark, on the right from "PYPI MANAGER" or use something like "i" button that would show (preferably non-blocking) informational popup/temporary window
3. (future) Develop a "Conda extension manager" and it will be used automatically in place of "PyPI extension manager" when conda env is detected. In jupyter-scheduler we implemented ["environment picker"](https://github.com/andrii-i/jupyter-scheduler/blob/main/src/components/environment-picker.tsx), maybe some code could be reused

**Comment by GabrielaVives** [comment]
Yes I think the first solution is good for a quick fix @andrii-i . I don't think this should be included in the `Warning` section, as it isn't related to the warning message. 

How do people use this information? do users need to copy/paste it somewhere or is it only descriptive information?

**Comment by krassowski** [comment]
The question mark icon is the absolute minimum as it does attract attention. No one will know to hover over "PYPI MANAGER".

Reducing visibility will lead to:
- **broken environments** (rarely) and
- (frequently) to changes in other dependencies (potentially corrupting the conclusions in the research process in the notebook).

This will lead to frustration, and some users will abandon JupyterLab after having such an experience. I had helped countless users on platforms like StackOverflow suffering from issues like that after installing incompatible packages. "Fixing" a problem by removing a visual indicator (which we discussed during a team meeting before, https://github.com/jupyterlab/jupyterlab/pull/12866#issuecomment-1225986122) does not seem right. In JupyterLab<4 extension manager was **never** modifying the Python/conda environments, this is a change in behaviour in 4.0 and I believe that we need to warn users; most of them will not read the release notes (nor will they notice the subtle change in visuals).

The ideal UX would be:
- environment name is shown visibly
- after pressing install:
   - nothing extra happens if the installation of the extension would be "clean"
   - if the installation of the extension modifies existing dependencies (e.g. has conflicting dependencies, or requires updating of installed packages), a confirmation dialog is shown with the list of changes and user needs to accept
   - if the PyPI manager were to install into a conda environment the confirmation dialog with appropriate message is always shown

> I don't think this should be included in the Warning section, as it isn't related to the warning message.

This is incorrect, circular reasoning. It is not related because we have not put it there yet. Once we explain that other dependencies may be updated when installing extension, and that this specific environment will be affected it will be related. Expanding upon scope beyond simply breaking integrity of environment and uncontrolled changes to package versions, if someone is using a remote kernel as a sandbox, they may expect that extensions will be installed in the kernel's environment **which is not the case**, exposing them to a security risk.

> do users need to copy/paste it somewhere or is it only descriptive information?

No, I think that in most cases users won't need to copy-paste it.

**Comment by andrii-i** [comment]
@krassowski Are we sure that people using JupyterLab know that you can hover over a question mark at all? I agree that information is important and that question mark attracts some attention but need to hover to get information is not clear at all. If this information is so important, this might be a problem in 4.0.0 release. Should we do a quick fix changing (?) to something like (i) and making it a clickable button that opens a popup?
